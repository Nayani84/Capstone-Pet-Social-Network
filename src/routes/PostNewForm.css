/* PostNewForm.css */
.PostNewForm-container {
    width: 100%;
    max-width: 600px;
    margin: 10px auto;
    padding: 20px;
    background-color: #f8f9fa;
    border: 1px solid #ddd;
    border-radius: 8px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
}

.PostNewForm-container h1 {
    text-align: center;
    margin-bottom: 20px;
}

.post-new-form {
    display: flex;
    flex-direction: column;
    gap: 15px;
}

.PostNewForm-container .form-group {
    display: flex;
    flex-direction: column;
    gap: 5px;
}

.PostNewForm-container input[type="text"] {
    padding: 10px;
    font-size: 1rem;
    border: 1px solid #ccc;
    border-radius: 4px;
    width: 100%;
    box-sizing: border-box;
    outline: none;
}

.PostNewForm-container .submit-button {
    padding: 12px 20px;
    background-color: #007bff;
    color: white;
    font-size: 1rem;
    font-weight: bold;
    text-align: center;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    transition: background-color 0.3s ease;
}

.PostNewForm-container .submit-button:hover {
    background-color: #0056b3;
}


/* Mobile responsiveness */
@media (max-width: 768px) {

    .post-new-form {
        gap: 10px;
    }

    .PostNewForm-container input[type="text"] {
        font-size: 0.9rem;
    }

    .PostNewForm-container .submit-button {
        font-size: 0.9rem;
        padding: 10px 15px;
    }

}