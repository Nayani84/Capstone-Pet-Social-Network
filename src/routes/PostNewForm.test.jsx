import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import PostNewForm from "./PostNewForm";
import { vi } from "vitest";
import PetApi from "../PetApi";

// Mocking the PetApi module
vi.mock("../PetApi", () => ({
  default: {
    createPost: vi.fn(() => Promise.resolve({ id: 1 }))
  }
}));

// Mocking the useNavigate hook
vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return {
    ...actual,
    useNavigate: () => vi.fn().mockImplementation(() => (path) => console.log(`Navigate to ${path}`)),
  };
});

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <PostNewForm currentUser={{ id: 1 }} />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <PostNewForm currentUser={{ id: 1 }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("submits the form with post data", async function () {
  const { getByLabelText, getByText } = render(
    <MemoryRouter>
      <PostNewForm currentUser={{ id: 1 }} />
    </MemoryRouter>
  );

  const contentInput = getByLabelText("Content :");
  const imageUrlInput = getByLabelText("Image URL :");
  const submitButton = getByText("Add Post");

  // Simulate user typing into the input fields
  fireEvent.change(contentInput, { target: { value: "New Post Content" } });
  fireEvent.change(imageUrlInput, { target: { value: "http://example.com/image.jpg" } });

  fireEvent.click(submitButton);

  // Wait for expectations to avoid act warnings
  await waitFor(() => {
    expect(PetApi.createPost).toHaveBeenCalledWith({
      content: "New Post Content",
      imageUrl: "http://example.com/image.jpg",
      ownerId: 1
    });
  });
});

