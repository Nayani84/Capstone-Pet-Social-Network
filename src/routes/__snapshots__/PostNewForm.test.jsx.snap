// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="PostNewForm-container"
  >
    <h1>
      Add a New Post
    </h1>
    <form
      class="post-new-form"
      role="form"
    >
      <div
        class="form-group"
      >
        <label
          for="content"
        >
          Content :
        </label>
        <input
          id="content"
          name="content"
          placeholder="Content"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          for="imageUrl"
        >
          Image URL :
        </label>
        <input
          id="imageUrl"
          name="imageUrl"
          placeholder="Image URL"
          type="text"
          value=""
        />
      </div>
      <button
        class="submit-button"
        type="submit"
      >
        Add Post
      </button>
    </form>
  </div>
</DocumentFragment>
`;
